sda-commons-server-auth/src/main/java/org/sdase/commons/server/opa/extension/OpaInputExtension.java

```diff
@@ -0,0 +1,84 @@
+package org.sdase.commons.server.opa.extension;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import javax.ws.rs.container.ContainerRequestContext;
+
+/**
+ * An extension to provide additional data to the {@link
+ * org.sdase.commons.server.opa.filter.model.OpaInput} before sending it to the Open Policy Agent.
+ *
+ * <p>Implementing this class should be the <i>last resort</i>. It is more favourable to depend on
+ * the existing {@link org.sdase.commons.server.opa.extension extensions} and instead use the
+ * constraint feature of the {@link
+ * org.sdase.commons.server.opa.OpaJwtPrincipal#getConstraintsAsEntity(Class)} to receive data from
+ * the policy decider and use it to decide based on it in your service.
+ *
+ * <p>Each extension is added in its own namespace, that means that it's data is accessible in a
+ * single sub-property. This contents can be represented by any object that is serializable by an
+ * {@link ObjectMapper}.
+ *
+ * <p>Example that returns a boolean that was serialized with the provided {@link ObjectMapper}:
+ *
+ * <pre>
+ *   {
+ *     "jwt": "…",
+ *     "path": ["…", "…"],
+ *     "httpMethod": "GET",
+ *     "myExtension": true
+ *   }
+ * </pre>
+ *
+ * <p>Example that returns an object that was serialized with the provided {@link ObjectMapper}:
+ *
+ * <pre>
+ *   {
+ *     "jwt": "…",
+ *     "path": ["…", "…"],
+ *     "httpMethod": "GET",
+ *     "myExtension": {
+ *       "myBoolean": true,
+ *       "myString": "asdf",
+ *       "myArray": ["…", "…", "…"]
+ *     }
+ *   }
+ * </pre>
+ *
+ * Please note that is is prohibited to override properties that are already set in the original
+ * {@link org.sdase.commons.server.opa.filter.model.OpaInput}.
+ */
+public interface OpaInputExtension {
+
+  /**
+   * The namespace (i.e. property name) where this extension is included. Please stick to naming the
+   * property in snake_case or camelCase.
+   *
+   * @return the namespace to store the value in.
+   */
+  String getNamespace();
```

Comment:
If we add the namespace at time of registration the consumers have more options and this interface can be a functional interface to be implemented on demand.